src/PatchOrchestrationApplication/NodeAgentNTService/src/Manager/WindowsUpdateManager.cs

```diff
@@ -223,6 +226,78 @@ private void StopFabricServices()
         }
 
+        /// <summary>
+        /// This method tries to read the last operation(download/install) start time saved in the file.
+        /// </summary>
+        private DateTime ReadLastOperationStartTimeStamp()
+        {
+            string LastUpdateOperationStartTimeStampFilePath = this.GetLastOperationStartTimeStampFilePath();
+            if (File.Exists(LastUpdateOperationStartTimeStampFilePath))
+            {
+                try
+                {
+                    string text = File.ReadAllText(LastUpdateOperationStartTimeStampFilePath).Trim();
+                    return DateTime.ParseExact(text, "yyyyMMddHHmmss", null);
+                }
+                catch(Exception ex)
+                {
+                    _eventSource.WarningMessage(string.Format("LastOperationStartTimeStamp parsing failed with execption : ex {0}", ex.ToString()));
+                }
+            }
+            return default(DateTime);
+        }
+
+        /// <summary>
+        /// This method tries to update the previous operation(download/install) start time saved in the file.
+        /// </summary>        
+        private void UpdateLastOperationStartTimeStamp(DateTime timeStamp)
+        {
+            try
+            {
+                this.WriteLastOperationStartTimeStamp(timeStamp);
+            }
+            catch(Exception ex)
+            {
+                this._eventSource.WarningMessage(string.Format("WriteLastOperationStartTimeStamp failed with exception {0}", ex.ToString()));
+                return;
+            }
+            this.lastUpdateOperationStartTimeStamp = timeStamp;
+        }
+
+        /// <summary>
+        /// This method tries to write the previous operation(download/install) start time saved in the file.
+        /// </summary>
+        private void WriteLastOperationStartTimeStamp(DateTime timeStamp)
+        {
+            string randomFilePath = Path.Combine(this._settingsManager.TempFolder, Path.GetRandomFileName());
+            if(File.Exists(randomFilePath))
+            {
+                File.Delete(randomFilePath);
+            }
+
+            using (FileStream fs = File.Create(randomFilePath))
+            {
+                string text = timeStamp.ToString("yyyyMMddHHmmss");
+                Byte[] info = new System.Text.UTF8Encoding(true).GetBytes(text);
+                fs.Write(info, 0, info.Length);
+            }
+
+            string lastOperationStartTimeStampFilePath = this.GetLastOperationStartTimeStampFilePath();
+
+            if (File.Exists(lastOperationStartTimeStampFilePath))
+            {
+                File.Delete(lastOperationStartTimeStampFilePath);
+            }
+            File.Move(randomFilePath, lastOperationStartTimeStampFilePath);
+            File.Delete(randomFilePath);
+            _eventSource.InfoMessage("LastOperationStartTimeStampFile written with timeStamp : {0}", this.lastUpdateOperationStartTimeStamp);
```

Comment:
>lastUpdateOperationStartTimeStamp [](start = 106, length = 33)

incorrect variable you are logging here.

Reply:
Thanks